### ISSUE TITLE
Undefined array key 3 in ./inc/media.php on line 1700

### ISSUE THREAD

**Issue by dregad** [dokuwiki/dokuwiki#4276]
### The problem

My wiki's error log is swamped with hundreds of occurrences of the following errors, every day.

E_WARNING: Undefined array key 3 in ./inc/media.php(1700)
E_WARNING: Trying to access array offset on value of type null in ./inc/media.php(1700)

I was initially not able to reproduce the problem, so I searched the web server's logs for more information and found out that most of these requests seem to be triggered by bots; analyzing the queries, I realized that the problem occurs when an unknown or unaccessible namespace is referenced when opening the media manager with the following URL

http://example.com/dokuwiki/doku.php?do=media&ns=XXX

Where XXX is the offending namespace (e.g. from my logs: `wiki:backup`, which does not exist)

### Version of DokuWiki

2024-02-06a Kaos (reproducible in master @ 31afae8a5238b24d15fc0e6f342ee8341d1014dd)

### PHP Version

8.2.16

**Comment by dregad** [comment]
> the problem occurs when an unknown or unaccessible namespace is referenced

This is not entirely true actually, something fishy is going on here...

Given the following namespaces hierarchy
```
+- root
   +- playground
   +- wiki
```

Accessing the URL http://example.com/dokuwiki/doku.php?do=media&ns=XXX

| Value of XXX | Result | Comments |
|---|---|---|
|aaa | ✅ | Non-existing namespace _aaa_ inserted before the first "real" namespace (_playground_) |
|zzz |  ❌  | _zzz_ namespace not listed in hierarchy, PHP warnings occur |
|qqq | ✅ | Non-existing namespace inserted between *playground* and *wiki* namespaces |
|playground | ✅ | _playground_ namespace selected |
|playground:xxx | ✅ | Non-existing sub-namespace _xxx_ inserted below _playground_ |
|wiki:xxx |  ❌  | Non-existing _wiki:xxx_ namespace not listed in hierarchy, PHP warnings occur |

**Comment by fiwswe** [comment]
> 2024-02-14a Kaos

Not very likely ;-) But let's assume you meant 2024-02-06a Kaos.

I can **not** reproduce this using:
`$ curl --url 'https://<myhostname>/doku.php?do=media&ns=wiki:xyz'`
where the namespace `wiki:xyz` does not exist. I also tried `aaa` and `wiki:xxx` (which also don't exist) with the same result. But I do have a more complicated hierarchy of namespaces in the wiki I was using.

DW: 2024-02-06a Kaos
PHP: 8.3.6 (using PHP-FPM)

So whatever is going on, must be still more obscure.

**Comment by dregad** [comment]
> > 2024-02-14a Kaos
> 
> Not very likely ;-) But let's assume you meant 2024-02-06a Kaos.

Indeed. Sorry about that, fixed.

> I can not reproduce this [...] But I do have a more complicated hierarchy of namespaces in the wiki I was using.

Do you have any namespaces after `wiki` (in lexicographical order) ? The key is the namespace specified by *ns* parameter, which must be higher than your last namespace's name (i.e. strcmp('XXX', 'yournamespace') > 0)

**Comment by dregad** [comment]
I think I've nailed it. I'll submit a pull request shortly.

**Comment by fiwswe** [comment]
> Do you have any namespaces after wiki (in lexicographical order) ? The key is the namespace specified by ns parameter, which must be higher than your last namespace's name (i.e. strcmp('XXX', 'yournamespace') > 0)

No. I don't have any sub-namespaces in the `wiki` ns and my last root ns starts with a `w` So I tried the same thing with `ns=xxx`. Alas same result.

Note: I am seeing some unrelated warnings from the Discussion plugin which I have installed (not actively used except for a few test pages). I don't think these warnings have any bearing on your issue though.
`PHP message: PHP Warning:  Undefined array key "comment" in /dokuwiki/lib/plugins/discussion/action.php on line 222`

**Comment by dregad** [comment]
I don't know what to say. This is consistently reproducible for me, with a fresh install of DokuWiki from Git master, with every plugin disabled.

**Comment by dregad** [comment]
OK let me rephrase that.

With master branch, I get the PHP warnings, but the temporary namespace is added.

![image](https://github.com/dokuwiki/dokuwiki/assets/449891/9e16d27d-e345-4811-8c81-a4a56882c26a)

I was actually testing with a commit fixing the PHP warning that started this whole thing (adding an array_key_exists() check to the while loop in media_nstree()). In this case, no more warnings but the temp namespace is gone.

![image](https://github.com/dokuwiki/dokuwiki/assets/449891/d4b3e114-0e85-4114-9882-275f0a5ea4ec)
